ui/views/tx/details/tx-add-signature-view.js

```diff
@@ -1,37 +1,33 @@
-import React, {useState} from 'react'
+import React, {useCallback, useState} from 'react'
 import {Button} from '@stellar-expert/ui-framework'
 import albedo from '@albedo-link/intent'
 import {submitTx} from '../../../infrastructure/tx-dispatcher'
 
-export default function TxAddSignatureView({txInfo, onUpdate, updateResult}) {
-    const [error, setError] = useState(''),
-        [inProgress, setInProgress] = useState(false)
+export default function TxAddSignatureView({txInfo, onUpdate}) {
+    const [error, setError] = useState('')
+    const [inProgress, setInProgress] = useState(false)
 
-    if (txInfo.readyToSubmit || txInfo.submitted) return null
-
-    function requestSignature(updateResult) {
+    const requestSignature = useCallback(() => {
         setError('')
         setInProgress(true)
-        updateResult(null)
         albedo.tx({xdr: txInfo.xdr, network: txInfo.network})
             .then(({signed_envelope_xdr: xdr}) => submitTx({...txInfo, xdr}))
-            .then(txInfo => {
-                updateResult(txInfo)
-                onUpdate(txInfo)
-            })
+            .then(txInfo => onUpdate(txInfo))
             .catch(e => {
                 setError(e.message)
             })
             .finally(() => setInProgress(false))
-    }
+    }, [txInfo, onUpdate])
+
+    if (txInfo.readyToSubmit || txInfo.submitted) return null
```

Comment:
```suggestion
    if (txInfo.readyToSubmit || txInfo.submitted)
      return null
```
Try to always move `return` and `throw` statements to the new line -- it's easier to set a breakpoint during debug in the future.